src/daft-context/src/python.rs

```diff
@@ -46,13 +49,85 @@ impl PyDaftContext {
     pub fn set_daft_planning_config(&self, py: Python, config: PyDaftPlanningConfig) {
         py.allow_threads(|| self.inner.set_planning_config(config.config));
     }
+
+    pub fn attach_subscriber(&self, py: Python, subscriber: PyObject) {
+        py.allow_threads(|| {
+            self.inner
+                .attach_subscriber(Arc::new(subscribers::python::PyQuerySubscriberWrapper(
+                    subscriber,
+                )));
+        });
+    }
+
+    pub fn notify_query_start(
```

Comment:
I may just expose an API for looking at the subscribers directly, haven't decided just yet.